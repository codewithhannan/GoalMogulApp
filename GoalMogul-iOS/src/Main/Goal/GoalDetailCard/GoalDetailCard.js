import React, { Component } from 'react';
import {
  View,
  Image,
  Text,
  ScrollView
} from 'react-native';
import { connect } from 'react-redux';
import { Icon } from 'react-native-elements';

// Component
import SearchBarHeader from '../../../Main/Common/Header/SearchBarHeader';
import SuggestionModal from './SuggestionModal';
import Headline from '../Common/Headline';
import Timestamp from '../Common/Timestamp';
import ActionButton from '../Common/ActionButton';
import ActionButtonGroup from '../Common/ActionButtonGroup';
import SectionCard from '../Common/SectionCard';
import TabButtonGroup from '../Common/TabButtonGroup';
import TabButton from '../Common/TabButton';
import ProgressBar from '../Common/ProgressBar';

// Asset
import defaultProfilePic from '../../../asset/utils/defaultUserProfile.png';
import LoveIcon from '../../../asset/utils/love.png';
import BulbIcon from '../../../asset/utils/bulb.png';
import ShareIcon from '../../../asset/utils/forward.png';
import HelpIcon from '../../../asset/utils/help.png';
import StepIcon from '../../../asset/utils/steps.png';

const testNeed = [
  {
    text: 'Get in contact with Nuclear expert'
  },
  {
    text: 'Introduction to someone from Bill and Melinda Gates foundation'
  },


];

const testStep = [
  {
    text: 'step 1'
  }
];

class GoalDetailCard extends Component {
  state = {
    // tab state stays within each component
    tab: 'comments',
    suggestionModal: true,
  }

  // Card central content. Progressbar for goal card
  renderCardContent() {
    return (
      <View style={{ marginTop: 20 }}>
        <ProgressBar startTime='Mar 2013' endTime='Nov 2011' />
      </View>
    );
  }

  // user basic information
  renderUserDetail() {
    return (
      <View style={{ flexDirection: 'row' }}>
        <Image source={defaultProfilePic} resizeMode='contain' style={{ height: 60, width: 60 }} />
        <View style={{ marginLeft: 15, flex: 1 }}>
          <Headline name='John Doe' category='Personal Development' />
          <Timestamp time='5 mins ago' />
          <View style={{ flexDirection: 'row', marginTop: 10 }}>
            <Text
              style={{ flex: 1, flexWrap: 'wrap', color: 'black', fontSize: 13 }}
              numberOfLines={3}
              ellipsizeMode='tail'
            >
              Establish a LMFBR near Westport, Connecticut by the year 2020
            </Text>
          </View>

        </View>
      </View>
    );
  }

  renderTabs() {
    return (
      <TabButtonGroup>
        <TabButton
          text='Comments'
          selected={this.state.tab === 'comments'}
          count={25}
          onPress={() => this.setState({ tab: 'comments' })}
        />
        <TabButton
          text='Needs and Steps'
          iconStyle={{ height: 20, width: 20 }}
          selected={this.state.tab === 'steps'}
          count={20}
          onPress={() => this.setState({ tab: 'steps' })}
        />
      </TabButtonGroup>
    );
  }

  renderSections() {
    const needs = this.renderNeeds();
    const steps = this.renderSteps();

    return (
      <ScrollView>
        {needs}
        {steps}
      </ScrollView>
    );
  }

  // Render needs
  renderNeeds() {
    if (!testNeed || testNeed.length === 0) {
      return;
    }
    const title = (
      <SectionTitle
        iconSource={HelpIcon}
        text='Needs'
        count={5}
      />
    );

    const needs = testNeed.map((need, index) => {
      return <SectionCard key={index} />;
    });

    return (
      <View>
        {title}
        {needs}
      </View>
    );
  }

  // Render steps
  renderSteps() {
    if (!testStep || testNeed.length === 0) {
      return;
    }
    const title = (
      <SectionTitle
        iconSource={StepIcon}
        iconStyle={{ height: 20, width: 20 }}
        text='Steps'
        count={7}
      />
    );

    const steps = testStep.map((step, index) => {
      return <SectionCard key={index} />;
    });

    return (
      <View>
        {title}
        {steps}
      </View>
    );
  }

  renderActionButtons() {
    return (
      <ActionButtonGroup>
        <ActionButton
          iconSource={LoveIcon}
          count={22}
          iconContainerStyle={{ backgroundColor: '#f9d6c9' }}
          iconStyle={{ tintColor: '#f15860' }}
          onPress={() => console.log('like')}
        />
        <ActionButton
          iconSource={ShareIcon}
          count={5}
          iconStyle={{ tintColor: '#a8e1a0', height: 32, width: 32 }}
          onPress={() => console.log('share')}
        />
        <ActionButton
          iconSource={BulbIcon}
          count={45}
          iconStyle={{ tintColor: '#f5eb6f', height: 26, width: 26 }}
          onPress={() => console.log('suggest')}
        />
      </ActionButtonGroup>
    );
  }

  render() {
    return (
      <View style={{ backgroundColor: '#e5e5e5', flex: 1 }}>
        <SuggestionModal
          visible={this.state.suggestionModal}
          onCancel={() => this.setState({ suggestionModal: false })}
        />
        <SearchBarHeader backButton title='Goal' />
        <View style={{ ...styles.containerStyle, marginTop: 2 }}>
          <View style={{ marginTop: 20, marginBottom: 10, marginRight: 15, marginLeft: 15 }}>
            {this.renderUserDetail()}
            {this.renderCardContent()}
          </View>
        </View>

        <View style={styles.containerStyle}>
          {this.renderActionButtons()}
        </View>

        {this.renderTabs()}
        {this.renderSections()}
      </View>
    );
  }
}

const styles = {
  containerStyle: {
    backgroundColor: 'white',
  },
  iconStyle: {
    alignSelf: 'center',
    fontSize: 20,
    marginLeft: 5,
    marginTop: 2
  },
  sectionTitleStyle: {
    containerStyle: {
      alignItems: 'center',
      flexDirection: 'row',
      height: 38,
      marginLeft: 15
    },
    iconStyle: {
      height: 26,
      width: 26,
      tintColor: '#616161'
    },
    textStyle: {
      fontSize: 11,
      marginLeft: 8,
      color: '#616161'
    },
    countTextStyle: {
      fontSize: 11,
      color: '#616161'
    }
  }
};

const SectionTitle = (props) => {
  const { sectionTitleStyle } = styles;
  const image = props.iconSource ?
    (<Image
      source={props.iconSource}
      style={{ ...sectionTitleStyle.iconStyle, ...props.iconStyle }}
    />)
    : '';

  return (
    <View style={{ ...sectionTitleStyle.containerStyle }}>
      {image}
      <Text style={{ ...sectionTitleStyle.textStyle, ...props.textStyle }}>
        {props.text}
      </Text>
      <Icon name='dot-single' type='entypo' color='#616161' size={20} />
      <Text style={sectionTitleStyle.countTextStyle}>{props.count}</Text>
    </View>
  );
};

export default connect(
  null,
  null
)(GoalDetailCard);
