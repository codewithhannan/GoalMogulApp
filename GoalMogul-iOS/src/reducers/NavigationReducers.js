import { ActionConst, Actions } from 'react-native-router-flux';
import _ from 'lodash';

export const initialState = {
  scene: {},
  stack: [],
  state: {},
  tab: undefined
};

// main navigation routes for the application
export const MAIN_NAVIGATION_ROUTES = ['homeTab', 'meetTab', 'notificationTab', 'exploreTab', 'chatTab'];

export default (state = initialState, action = {}) => {
  switch (action.type) {
    // focus action is dispatched when a new screen comes into focus
    // case ActionConst.FOCUS: {
    //   return {
    //     ...state,
    //     scene: action.scene,
    //     state: Actions.state
    //   };
    // }
    //
    // case ActionConst.POP: {
    //   const newStack = { ...state.stack };
    //   newStack.pop();
    //   return { ...state, stack: newStack, state: Actions.state };
    // }

    // case ActionConst.PUSH: {
      // console.log('push new scene', action);
      // const newStack = [...state.stack];
      // console.log('new stack is: ', newStack);
      // newStack.push(action.routeName);
      // return { ...state, stack: newStack };
    // }

    case 'Navigation/NAVIGATE': {
      // console.log('navigate to: ', action.routeName);
      const newState = _.cloneDeep(state);
      const { routeName } = action;
      if (!routeName) return newState;

      let route;
      if (routeName === 'mainTabs') route = 'homeTab';
      if (!MAIN_NAVIGATION_ROUTES.some((r) => r === routeName)) {
        return newState;
      }
      route = routeName;
      return _.set(newState, 'tab', route);
    }

    default:
      return { ...state };
  }
};
