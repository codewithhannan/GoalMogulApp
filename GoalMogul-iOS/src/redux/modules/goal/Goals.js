/**
 * This reducer is the source of truth for Goals related Components based on the design
 * https://docs.google.com/document/d/1JxjCLZGsp5x_Sy0R56y05aUgAUALtuSVAgJIway9JTg/edit?usp=sharing
 */

import _ from 'lodash';

// Constants
import {
    PROFILE_FETCH_TAB_DONE,
    PROFILE_REFRESH_TAB_DONE,
    PROFILE_GOAL_DELETE_SUCCESS
} from '../../../reducers/Profile';

import {
    PROFILE_CLOSE_PROFILE
} from '../../../actions/types';

/**
 * List of const to add 
 * 
 * Create Goal
 * GOAL_CREATE_EDIT_SUCCESS
 * 
 * Goal Detail related
 * GOAL_DETAIL_MARK_NEED_AS_COMPLETE_SUCCESS
 * GOAL_DETAIL_MARK_AS_COMPLETE_SUCCESS,
 * GOAL_DETAIL_SHARE_TO_MASTERMIND_SUCCESS,
 * GOAL_DETAIL_MARK_STEP_AS_COMPLETE_SUCCESS,
 * GOAL_DETAIL_MARK_NEED_AS_COMPLETE_SUCCESS
 * 
 * Profile related (done)
 * PROFILE_FETCH_TAB_DONE (done)
 * PROFILE_REFRESH_TAB_DONE (done)
 * PROFILE_GOAL_DELETE_SUCCESS (done)
 * PROFILE_CLOSE_PROFILE (done)
 * 
 * Comment related
 * The ones that need to increase / decrease comment count
 * 
 * Home related (Goal Feed)
 * 
 */

const DEBUG_KEY = '[ Reducer Goals ]';

// Sample goal object in the map
const INITIAL_GOAL = {
    goal: {},
    pageId: {
        refreshing: false
    },
    reference: [],
};

const INITIAL_GOAL_PAGE = {
    refreshing: false,
    // Potential navigation state and etc. First focus on integration with Profile
};

const INITIAL_STATE = {

};

export default (state = INITIAL_STATE, action) => {
    switch (action.type) {
        /* Profile related */
        case PROFILE_REFRESH_TAB_DONE:
        case PROFILE_FETCH_TAB_DONE: {
            const { pageId, data, type } = action.payload;
            let newState = _.cloneDeep(state);
            if (type !== 'goals') return newState;
            if (!data || _.isEmpty(data)) return newState;

            data.forEach(goal => {
                const goalId = goal._id;
                // Update goal
                if (!_.has(newState, goalId)) {
                    newState = _.set(newState, `${goalId}.goal`, goal);
                }

                const oldReference = _.get(newState, `${goalId}.reference`);
                const hasPageReference = (oldReference !== undefined && oldReference.some(r => r === pageId));
                // Update reference
                let newReference = [pageId];
                if (!hasPageReference) {
                    newReference = newReference.concat(oldReference);
                }
                newState = _.set(newState, `${goalId}.reference`, newReference);
            });

            return newState;
        }

        case PROFILE_GOAL_DELETE_SUCCESS: {
            const { pageId, goalId } = action.payload;
            let newState = _.cloneDeep(state);
            if (!_.has(newState, goalId)) return newState; // no reference to remove

            let goalToUpdate = _.get(newState, `${goalId}`);

            // Update reference
            const oldReference = _.get(goalToUpdate, 'reference');
            let newReference = oldReference;
            if (oldReference && oldReference.some(r => r === pageId)) {
                newReference = newReference.filter(r => r !== pageId);
            }

            // Remove pageId reference object
            goalToUpdate = _.omit(goalToUpdate, `${pageId}`);

            // Remove this goal if it's no longer referenced
            if (!newReference || _.isEmpty(newReference)) {
                newState = _.omit(newState, `${goalId}`);
                return newState;
            }

            // Update the goal by goalId
            goalToUpdate = _.set(goalToUpdate, 'reference', newReference);
            newState = _.set(newState, `${goalId}`, goalToUpdate);
            return newState;
        }

        case PROFILE_CLOSE_PROFILE: {
            const { goalList, pageId } = action.payload;
            let newState = _.cloneDeep(state);
            if (!goalList || _.isEmpty(goalList)) return newState;

            goalList.forEach(goalId => {
                // Check if goalId in the Goals
                if (!_.has(newState, goalId)) {
                    return;
                }

                let goalToUpdate = _.get(newState, `${goalId}`);

                // Remove pageId reference object
                goalToUpdate = _.omit(goalToUpdate, `${pageId}`);

                const oldReference = _.get(newState, `${goalId}.reference`);
                const hasPageReference = (oldReference !== undefined && oldReference.some(r => r === pageId));
                // Remove reference
                let newReference = oldReference;
                if (hasPageReference) {
                    newReference = oldReference.filter(r => r !== pageId);
                }

                // Remove this goal if it's no longer referenced
                if (!newReference || _.isEmpty(newReference)) {
                    newState = _.omit(newState, `${goalId}`);
                    return;
                }
                goalToUpdate = _.set(goalToUpdate, 'reference', newReference);
                newState = _.set(newState, `${goalId}`, goalToUpdate);
            });

            return newState;
        }

        default: 
            return { ...state };
    }
};
