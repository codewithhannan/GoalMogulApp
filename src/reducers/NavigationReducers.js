/** @format */

import _ from 'lodash'

export const initialState = {
    scene: {},
    stack: [],
    state: {},
    tab: undefined,
}

// main navigation routes for the application
export const MAIN_NAVIGATION_ROUTES = [
    'homeTab',
    'profileTab',
    'notificationTab',
    'exploreTab',
    'chatTab',
]

export default (state = initialState, action = {}) => {
    switch (action.type) {
        case 'Navigation/NAVIGATE': {
            const newState = _.cloneDeep(state)
            const { routeName } = action
            if (!routeName) return newState

            let route
            if (routeName === 'mainTabs') route = 'homeTab'
            if (!MAIN_NAVIGATION_ROUTES.some((r) => r === routeName)) {
                return newState
            }
            route = routeName
            return _.set(newState, 'tab', route)
        }

        default:
            return { ...state }
    }
}
