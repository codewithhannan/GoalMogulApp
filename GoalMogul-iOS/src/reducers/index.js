import { combineReducers } from 'redux';
import storage from 'redux-persist/lib/storage';
import { persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { reducer as formReducer } from 'redux-form';

/* reducers */
import AuthReducers from './AuthReducers';
import RegReducers from './RegReducers';
import CameraRollReducers from './CameraRollReducers';
import User from './User';
import Profile from './Profile';
import Setting from './Setting';
import NavigationReducers from './NavigationReducers';
import MeetReducers from './MeetReducers';
import Home from './Home';
import Search from '../redux/modules/search/Search';
import GoalDetailReducers from './GoalDetailReducers';
import CreateGoal from '../redux/modules/goal/CreateGoal';
import PostReducers from '../redux/modules/feed/post/PostReducers';
import ShareReducers from '../redux/modules/feed/post/ShareReducers';
import TribeReducers from '../redux/modules/tribe/TribeReducers';
import EventReducers from '../redux/modules/event/EventReducers';
import CommentReducers from '../redux/modules/feed/comment/CommentReducers';
import NewCommentReducers from '../redux/modules/feed/comment/NewCommentReducers';
import SuggestionSearchReducers from '../redux/modules/feed/comment/SuggestionSearchReducers';
import ReportReducers from '../redux/modules/report/ReportReducers';
import NotificationTabReducers from '../redux/modules/notification/NotificationTabReducers';
// Explore tab related reducers
import EventTabReducers from '../redux/modules/event/EventTabReducers';
import TribeTabReducers from '../redux/modules/tribe/TribeTabReducers';
import ExploreReducers from '../redux/modules/explore/ExploreReducers';
import NewShareReducers from '../redux/modules/feed/post/NewShareReducers';
import NewTribeReducers from '../redux/modules/tribe/NewTribeReducers';
import NewEventReducers from '../redux/modules/event/NewEventReducers';
// Menu related reducers
import MyEventTabReducers from '../redux/modules/event/MyEventTabReducers';
import MyEventReducers from '../redux/modules/event/MyEventReducers';
import MyTribeTabReducers from '../redux/modules/tribe/MyTribeTabReducers';
import MyTribeReducers from '../redux/modules/tribe/MyTribeReducers';
// Chat related reducers
import ChatTabReducers from '../redux/modules/chat/ChatTabReducers';

const rootPersistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2, // see "Merge Process" section for details.
  blacklist: ['registration', 'form']
};

const reducers = combineReducers({
  auth: AuthReducers,
  registration: RegReducers,
  cameraRoll: CameraRollReducers,
  user: User,
  profile: Profile,
  form: formReducer,
  setting: Setting,
  navigation: NavigationReducers,
  meet: MeetReducers,
  home: Home,
  search: Search,
  goalDetail: GoalDetailReducers,
  createGoal: CreateGoal,
  postDetail: PostReducers,
  shareDetail: ShareReducers,
  tribe: TribeReducers,
  event: EventReducers,
  comment: CommentReducers,
  newComment: NewCommentReducers,
  report: ReportReducers,
  explore: ExploreReducers,
  eventTab: EventTabReducers,
  tribeTab: TribeTabReducers,
  newShare: NewShareReducers,
  suggestionSearch: SuggestionSearchReducers,
  // menu related reducers. The following four reducers can be abstracted out later
  myEventTab: MyEventTabReducers,
  myTribeTab: MyTribeTabReducers,
  myEvent: MyEventReducers,
  myTribe: MyTribeReducers,
  newTribe: NewTribeReducers,
  newEvent: NewEventReducers,
  notification: NotificationTabReducers,
  chat: ChatTabReducers
});

export default persistReducer(rootPersistConfig, reducers);
