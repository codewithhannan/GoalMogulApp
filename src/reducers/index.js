/** @format */

import { combineReducers } from 'redux'
import storage from 'redux-persist/lib/storage'
import { persistReducer } from 'redux-persist'
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import { reducer as formReducer } from 'redux-form'

/* reducers */
import Challenges from './Challenges'
import AuthReducers from './AuthReducers'
import RegReducers from './RegReducers'
import GoalPrivacyReducers from './GoalPrivacy'
import CameraRollReducers from './CameraRollReducers'
import User from './User'
import Users from '../redux/modules/User/Users'
import Profile from './Profile'
import Setting from './Setting'
import NavigationReducers from './NavigationReducers'
import MeetReducers from './MeetReducers'
import Home from './Home'
import Search from '../redux/modules/search/Search'
import GoalDetailReducers from './GoalDetailReducers'
import Goals from '../redux/modules/goal/Goals'
import CreateGoal from '../redux/modules/goal/CreateGoal'
import PostReducers from '../redux/modules/feed/post/PostReducers'
import Posts from '../redux/modules/feed/post/Posts'
import ShareReducers from '../redux/modules/feed/post/ShareReducers'
import Tribes from '../redux/modules/tribe/Tribes'
import EventReducers from '../redux/modules/event/EventReducers'
import Events from '../redux/modules/event/Events'
import CommentReducers from '../redux/modules/feed/comment/CommentReducers'
import Comments from '../redux/modules/feed/comment/Comments'
import NewCommentReducers from '../redux/modules/feed/comment/NewCommentReducers'
import SuggestionSearchReducers from '../redux/modules/feed/comment/SuggestionSearchReducers'
import ReportReducers from '../redux/modules/report/ReportReducers'
import NotificationTabReducers from '../redux/modules/notification/NotificationTabReducers'
// Explore tab related reducers
import EventTabReducers from '../redux/modules/event/EventTabReducers'
import TribeTabReducers from '../redux/modules/tribe/TribeTabReducers'
import ExploreReducers from '../redux/modules/explore/ExploreReducers'
import NewShareReducers from '../redux/modules/feed/post/NewShareReducers'
import NewTribeReducers from '../redux/modules/tribe/NewTribeReducers'
import NewEventReducers from '../redux/modules/event/NewEventReducers'
// Menu related reducers
import MyEventTabReducers from '../redux/modules/event/MyEventTabReducers'
import MyEventReducers from '../redux/modules/event/MyEventReducers'
import MyTribeTabReducers from '../redux/modules/tribe/MyTribeTabReducers'
// Chat related reducers
import ChatReducers from '../redux/modules/chat/ChatReducers'
import CreateChatRoomReducers from '../redux/modules/chat/CreateChatRoomReducers'
import ChatRoomReducers from '../redux/modules/chat/ChatRoomReducers'
import TabIconReducers from '../redux/modules/navigation/TabIconReducers'
import ChatRoomMembersReducers from '../redux/modules/chat/ChatRoomMembersReducers'
import ShareToChatReducers from '../redux/modules/chat/ShareToChatReducers'
// Step by Step tutorial reducers
import Tutorials from '../redux/modules/User/Tutorials'
// Contact Sync
import ContactSyncReducers from '../redux/modules/User/ContactSync/ContactSyncReducers'

const rootPersistConfig = {
    key: 'root',
    storage,
    stateReconciler: autoMergeLevel2, // see "Merge Process" section for details.
    blacklist: ['registration', 'form'],
}

export const reducers = combineReducers({
    auth: AuthReducers,
    registration: RegReducers,
    cameraRoll: CameraRollReducers,
    user: User,
    users: Users,
    profile: Profile,
    form: formReducer,
    setting: Setting,
    navigation: NavigationReducers,
    meet: MeetReducers,
    home: Home,
    search: Search,
    goalDetail: GoalDetailReducers,
    goals: Goals,
    createGoal: CreateGoal,
    postDetail: PostReducers,
    posts: Posts,
    shareDetail: ShareReducers,
    tribes: Tribes,
    event: EventReducers,
    events: Events,
    comment: CommentReducers,
    newComment: NewCommentReducers,
    comments: Comments,
    report: ReportReducers,
    explore: ExploreReducers,
    eventTab: EventTabReducers,
    tribeTab: TribeTabReducers,
    newShare: NewShareReducers,
    suggestionSearch: SuggestionSearchReducers,
    // menu related reducers. The following four reducers can be abstracted out later
    myEventTab: MyEventTabReducers,
    myTribeTab: MyTribeTabReducers,
    myEvent: MyEventReducers,
    newTribe: NewTribeReducers,
    newEvent: NewEventReducers,
    newChatRoom: CreateChatRoomReducers,
    shareToChat: ShareToChatReducers,
    notification: NotificationTabReducers,
    chat: ChatReducers,
    chatRoom: ChatRoomReducers,
    chatRoomMembers: ChatRoomMembersReducers,
    navigationTabBadging: TabIconReducers,
    tutorials: Tutorials,
    contactSync: ContactSyncReducers,
    messageDoc: MessageDocReducers,
    challenges: Challenges,
    goalPrivacy: GoalPrivacyReducers,
})

export default persistReducer(rootPersistConfig, reducers)
