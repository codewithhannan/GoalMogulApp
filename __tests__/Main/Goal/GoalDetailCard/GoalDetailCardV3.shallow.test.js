/** @format */

import React from 'react'
import _ from 'lodash'
import { mount, shallow } from 'enzyme'
import { GoalDetailCardV3 } from '../../../../src/Main/Goal/GoalDetailCard/GoalDetailCardV3'
import { DEFAULT_GOAL } from '../../../__mocks__/MockGoals'
import { INITIAL_NAVIGATION_STATE_V2 } from '../../../../src/redux/modules/goal/Goals'
import { MAIN_NAVIGATION_ROUTES } from '../../../../src/reducers/NavigationReducers'

/**
 * This is a component test for GoalDetailCardV3. It shallow renders the page with
 * required props and expect the function to be called and state to be changed.
 *
 * The tests will manually update the props and pass into the component to verify
 * the necessary change.
 *
 * There are two parts of the tests.
 * First part is to very mapStateToProps which makes sure that redux states are
 * properly translated to props which are used by the component.
 *
 * Second part has three steps for each test
 * 1. specific props renders component with a specific state
 * 2. specific actions in component that call certain functions.
 * 3. after specific functions are triggered, we manually update the props to re-render the component
 *    and verify the component rendering
 */

describe('goal detail v3', () => {
    let component
    it('should shallow render basic goal detail v3 page', () => {
        component = shallow(<GoalDetailCardV3 {...DEFAULT_GOALDETAIL_PROPS} />)
        expect(component.debug()).toMatchSnapshot()
        expect(component.find('Spinner').exists()).toBeTruthy()
    })
})

const DEFAULT_GOALDETAIL_FUNCTIONS = {
    closeGoalDetail: jest.fn(),
    closeGoalDetailWithoutPoping: jest.fn(),
    attachSuggestion: jest.fn(),
    cancelSuggestion: jest.fn(),
    openSuggestionModal: jest.fn(),
    goalDetailSwitchTabV2: jest.fn(),
    goalDetailSwitchTabV2ByKey: jest.fn(),
    removeSuggestion: jest.fn(),
    createCommentFromSuggestion: jest.fn(),
    resetCommentType: jest.fn(),
    updateNewComment: jest.fn(),
    createCommentForSuggestion: jest.fn(),
    createSuggestion: jest.fn(),
    editGoal: jest.fn(),
    markGoalAsComplete: jest.fn(),
    refreshGoalDetailById: jest.fn(),
    refreshComments: jest.fn(),
    markUserViewGoal: jest.fn(),
    // Tutorial related
    showNextTutorialPage: jest.fn(),
    startTutorial: jest.fn(),
    updateNextStepNumber: jest.fn(),
    pauseTutorial: jest.fn(),
    saveTutorialState: jest.fn(),
}

const DEFAULT_GOALDETAIL_DATA = {
    commentLoading: false,
    stepsAndNeeds: [],
    comments: [],
    originalComments: [], // All comments in raw form
    goalDetail: DEFAULT_GOAL,
    navigationState: _.cloneDeep(INITIAL_NAVIGATION_STATE_V2),
    showingModalInDetail: false, // Can be potentially cleaned up
    showSuggestionModal: false,
    isSelf: true,
    tab: MAIN_NAVIGATION_ROUTES[0], // Default set to homeTab
    // When on focusTab, show the count for focusedItem
    focusedItemCount: 0,
    updating: false,
    newComment: undefined,
    // Tutorial related
    tutorialText: '',
    hasShown: true,
    showTutorial: false,
    copilotEvents: {
        on: jest.fn(),
        off: jest.fn(),
    },
}

const DEFAULT_GOALDETAIL_PROPS = {
    ...DEFAULT_GOALDETAIL_FUNCTIONS,
    ...DEFAULT_GOALDETAIL_DATA,
}
