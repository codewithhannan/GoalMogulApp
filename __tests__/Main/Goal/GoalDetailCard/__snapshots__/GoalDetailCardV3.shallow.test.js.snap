// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goal detail v3 should shallow render basic goal detail v3 page 1`] = `
"<MenuProvider customStyles={{...}} backHandler={false}>
  <View style={{...}}>
    <Spinner visible={false} customIndicator={{...}} cancelable={false} textContent=\\"\\" animation=\\"none\\" color=\\"white\\" size=\\"large\\" overlayColor=\\"rgba(0, 0, 0, 0.25)\\" />
    <Connect(SearchBarHeader) backButton={true} title=\\"Goal\\" onBackPress={[Function: onBackPress]} />
    <TabView style={{...}} navigationState={{...}} renderScene={[Function: bound ]} renderTabBar={[Function: bound ]} initialLayout={{...}} onIndexChange={[Function: onIndexChange]} swipeEnabled={false} tabBarPosition=\\"top\\" renderLazyPlaceholder={[Function: renderLazyPlaceholder]} keyboardDismissMode=\\"auto\\" lazy={false} lazyPreloadDistance={0} removeClippedSubviews={false} springConfig={{...}} timingConfig={{...}} gestureHandlerProps={{...}} renderPager={[Function: renderPager]} />
    <Connect(SuggestionModal) visible={false} onCancel={[Function: onCancel]} onAttach={[Function: onAttach]} pageId={[undefined]} goalId={[undefined]} item={{...}} />
  </View>
</MenuProvider>"
`;
